core/CMakeLists.txt

```diff
@@ -1,100 +1,100 @@
 file(GLOB_RECURSE KNOTING_SHADERS LIST_DIRECTORIES false "${CMAKE_SOURCE_DIR}/res/shaders/**/*.sc")
 
 file(GLOB_RECURSE KNOTING_SOURCES
-    src/*.cpp
-    inc/**/*.h
-)
+        src/*.cpp
+        inc/**/*.h
+        )
 
 set(PUBLIC_LIBS
-    cereal
-    glm
-    spdlog
-    imgui
-    imguizmo
-    EnTT::EnTT
-    stduuid
-    ${PHYSX_LIBRARIES}
-    bgfx
-    bx
-    bimg
-    rapidjson
-)
+        cereal
+        glm
+        spdlog
+        imgui
+        imguizmo
+        EnTT::EnTT
+        stduuid
+        ${PHYSX_LIBRARIES}
+        bgfx
+        bx
+        bimg
+        rapidjson
+        )
 
 set(PRIVATE_LIBS
-    glfw
-    stb
-)
+        glfw
+        stb
+        )
 
-add_library(knoting ${KNOTING_SOURCES} ${KNOTING_SHADERS})
```

Comment:
In the option to create files inside CLion you need to uncheck the option to add them to the CMakeFiles